homeassistant/components/verisure/switch.py

```diff
@@ -79,14 +81,14 @@ def available(self) -> bool:
 
     def turn_on(self, **kwargs: Any) -> None:
         """Set smartplug status on."""
-        self.coordinator.verisure.set_smartplug_state(self.serial_number, True)
+        self.coordinator.verisure.set_smartplug(self.serial_number, True)
```

Comment:
Tested this against the smart plug, but nothing happens.

The state is correctly reflected though.

Reply:
@persandstrom Can you assist? I don't have a smartplug to work with.

Reply:
Ran in through the debugger. Nothing is raised. The method call is handled as expected, the device label in the method call matches the actual device. 🤷

Reply:
The issue is that ```self.coordinator.verisure.set_smartplug(self.serial_number, True))``` is creating the query, not making the request.

 I see that this is not intuitive, and I will put some effort on the Session API. But for now just make the request:
 
 ```python
        self.coordinator.verisure.request(
            self.coordinator.verisure.set_smartplug(self.serial_number, True))
        self._state = True
```

Reply:
I can confirm that @persandstrom fix works!

Reply:
Does the `set_smartplug` request return a transaction_id, similar to  `door_lock` and `door_unlock`?

Reply:
@frenck @tobias-berg
Can you test it again please -> https://github.com/home-assistant/core/pull/89318/commits/ad7e6ac1f3870fee38fc5710026a229e527be1d7 ?